packages/babel-plugin-proposal-nullish-coalescing-operator/src/index.js

```diff
@@ -16,14 +16,17 @@ export default declare((api, { loose = false }) => {
           return;
         }
 
-        const ref = scope.generateUidIdentifierBasedOnNode(node.left);
-        scope.push({ id: ref });
+        let ref, assignment;
+        // skip creating extra reference when `left` is semantically safe to re-use
+        if (t.isIdentifier(node.left)) {
```

Comment:
What about using `isConstantExpression`, or maybe better `maybeGenerateMemoised`?

Reply:
Good idea! I end up with `maybeGenerateMemoised` because it is rare that the `left` is a literal --even if it is, creating a reference other than inlining this literal should costs less bytes in most situations, given that we generate `[_]+` as identifier name for literals.